### PR TITLE
504 docstring formatting

### PR DESCRIPTION
**Pull request by coquelin77**
## Description

New docstrings with custom formatting! major changes to almost every function. Special shoutout to @Cdebus for the organization and heavy lifting for these changes!

Issue/s resolved: #504

## Changes proposed:
- new docstring format
- new docstring rendering
- bugfix in sort

## Type of change
- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Documentation update

## Due Diligence

- [x] All split configurations tested
- [x] Multiple dtypes tested in relevant functions
- [x] Documentation updated
- [ ] Updated changelog.md under the title "Pending Additions"

#### Does this change modify the behaviour of other functions? If so, which?
no

### CODE REVIEW

**Comment by mtar** [comment]
restart tests

**Comment by mtar** [comment]
run tests

**Comment by coquelin77** [comment]
rerun tests

**Comment by mtar** [comment]
run tests

**Comment by mtar** [comment]
run tests

**Comment by mtar** [comment]
rerun tests

**Comment by mtar** [comment]
run tests

**Comment by mtar** [comment]
run tests

**Comment by mtar** [comment]
run tests

**Comment by mtar** [comment]
rerun tests

**Comment by mtar** [comment]
rerun tests

**Comment by mtar** [comment]
run tests

**Comment by coquelin77** [comment]
rerun tests

**Comment by coquelin77** [comment]
rerun tests

File: setup.py [hunk 1]
```diff
@@ -40,8 +39,9 @@
         "torchvision>=0.5.0",
```
Comment (mtar) [inline setup.py:39]: outdated?

Comment (coquelin77) [reply setup.py:39]: i think that torch would supersede this anyway, but it can be updated as well

Comment (coquelin77) [reply setup.py]: i updated it to match the requirements of torch 1.7

Comment (mtar) [reply setup.py]: I mean all listed requirements. Many of them are not working with python3.7+. Only the last line came up here.

Comment (coquelin77) [reply setup.py]: which ones dont work with 3.7?

Comment (mtar) [reply setup.py]: numpy scipy mpi4py pillow

Comment (coquelin77) [reply setup.py]: your package manager should take care of which is the best implementation which fits these requirements for your python version

**Review by Cdebus — APPROVED** [review]
I checked the rendering, looks good. We are ready for production

**Pull request merged**